src/persistentSubscription/getPersistentSubscriptionInfo.ts

```diff
@@ -0,0 +1,130 @@
+import { StreamIdentifier } from "../../generated/shared_pb";
+import { GetInfoReq } from "../../generated/persistent_pb";
+import {
+  PersistentSubscriptionsClient,
+  PersistentSubscriptionsService,
+} from "../../generated/persistent_grpc_pb";
+
+import {
+  convertToCommandError,
+  debug,
+  PersistentSubscriptionDoesNotExistError,
+} from "../utils";
+import { BaseOptions } from "../types";
+import { Client } from "../Client";
+import {
+  HTTPSubscriptionInfo,
+  mapHTTPPersistentSubscriptionInfo,
+  mapPersistentSubscriptionInfo,
+  PersistentSubscriptionInfo,
+} from "./utils/mapPersistentSubscriptionInfo";
+
+export interface GetPersistentSubscriptionInfoOptions extends BaseOptions {}
+
+declare module "../Client" {
+  interface Client {
+    /**
+     * Gets information and statistics on the specified persistent subscription and its connections.
+     * @param streamName A stream name.
+     * @param groupName A group name.
+     * @param options Replay options.
+     */
+    getPersistentSubscriptionInfo(
+      streamName: string,
+      groupName: string,
+      options?: GetPersistentSubscriptionInfoOptions
+    ): Promise<PersistentSubscriptionInfo>;
+  }
+}
+
+Client.prototype.getPersistentSubscriptionInfo = async function (
+  this: Client,
+  streamName: string,
+  groupName: string,
+  options: GetPersistentSubscriptionInfoOptions
+): Promise<PersistentSubscriptionInfo> {
+  debug.command("getPersistentSubscriptionInfo: %O", {
+    streamName,
+    groupName,
+    options,
+  });
+
+  if (await this.supports(PersistentSubscriptionsService.getInfo, "stream")) {
+    return getPersistentSubscriptionInfoGRPC.call(
+      this,
+      streamName,
+      groupName,
+      options
+    );
+  }
+
+  return getPersistentSubscriptionInfoHTTP.call(
+    this,
+    streamName,
+    groupName,
+    options
+  );
+};
+
+const getPersistentSubscriptionInfoGRPC = async function (
+  this: Client,
+  streamName: string,
+  groupName: string,
+  baseOptions: GetPersistentSubscriptionInfoOptions = {}
+): Promise<PersistentSubscriptionInfo> {
+  const req = new GetInfoReq();
+  const options = new GetInfoReq.Options();
+  const identifier = new StreamIdentifier();
+  identifier.setStreamName(Uint8Array.from(Buffer.from(streamName, "utf8")));
+  options.setStreamIdentifier(identifier);
```

Comment:
**SUGGESTION:** I see that this code is repeated in multiple places. It might be worth considering adding a factory method or other wrapper for that initialisation.

Reply:
Agreed: #278